sklearn/utils/testing.py

```diff
@@ -137,6 +137,7 @@ def assert_warns(warning_class, func, *args, **kw):
 
     """
     # very important to avoid uncontrolled state propagation
+    warnings.resetwarnings()
```

Comment:
Elsewhere there was a suggestion that we should never need to resetwarnings.

Reply:
Ah actually looking at the diff, `warnings.resetwarnings` has been removed from `clean_warning_registry` which is the change I was advocating.

It would be great if you could give us some reason why you need `warnings.resetwarnings` here. I am hoping we don't need to use it anywhere. In case we do need to use `warnings.resetwarnings`, it should be used inside a `warnings.catch_warnings` context manager so that the global `warnings.filters` is not affected.

Reply:
I commented somewhere else that you should remove these two lines:
```py
# Test that the warning registry is empty after assert_warns
assert_equal(sys.modules['warnings'].filters, [])
```

Expecting a assert_warns call to affect the global warnings.filters is not something I would expect. I can't see a good reason for it.

Reply:
where was the log? is it `./build_tools/circle/build_doc.sh`?